browser/brave_shields/ad_block_pref_service_factory.cc

```diff
@@ -35,9 +36,11 @@ AdBlockPrefServiceFactory::~AdBlockPrefServiceFactory() {}
 
 KeyedService* AdBlockPrefServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* context) const {
+  Profile* profile = Profile::FromBrowserContext(context);
   return new AdBlockPrefService(
-      g_brave_browser_process->ad_block_service(),
-      Profile::FromBrowserContext(context)->GetPrefs());
+      g_brave_browser_process->ad_block_service(), profile->GetPrefs(),
+      HostContentSettingsMapFactory::GetForProfile(profile),
+      profile->IsRegularProfile() && !profile->IsGuestSession());
```

Comment:
i guess GuestSession is never Regular (see comments for `IsRegularProfile()`)

Reply:
In general I'm not in favor of passing new things to this service just for P3A

Reply:
agreed, removed